packages/flutter/lib/src/cupertino/search_field.dart

```diff
@@ -0,0 +1,403 @@
+  /// `UISearchTextField` prefix look. The inset values were determined using
+  /// the comparison tool in https://github.com/flutter/platform_tests/.
+  final EdgeInsetsGeometry suffixInsets;
+
+  /// Sets the suffix widget's icon.
+  ///
+  /// Cannot be null. Defaults to the X-Mark [CupertinoIcons.xmark_circle_fill].
+  /// "To change the functionality of the suffix icon, provide a custom
+  /// onSuffixTap callback and specify an intuitive suffixIcon.
+  final Icon suffixIcon;
+
+  /// Dictates when the X-Mark (suffix) should be visible.
+  ///
+  /// Cannot be null. Defaults to only on when editing.
+  final OverlayVisibilityMode suffixMode;
+
+  /// Sets the X-Mark (suffix) action.
+  ///
+  /// Defaults to clearing the text. The suffix action is customizable
+  /// so that users can override it with other functionality, that isn't
+  /// necessarily clearing text.
+  final VoidCallback? onSuffixTap;
+
+  /// {@macro flutter.material.textfield.restorationId}
+  final String? restorationId;
+
+  /// {@macro flutter.widgets.Focus.focusNode}
+  final FocusNode? focusNode;
+
+  @override
+  State<StatefulWidget> createState() => _CupertinoSearchTextFieldState();
+}
+
+class _CupertinoSearchTextFieldState extends State<CupertinoSearchTextField>
+    with RestorationMixin {
+  /// Default value for the border radius. Radius value was determined using the
+  /// comparison tool in https://github.com/flutter/platform_tests/.
+  final BorderRadius _kDefaultBorderRadius =
+      const BorderRadius.all(Radius.circular(9.0));
+
+  RestorableTextEditingController? _controller;
+
+  TextEditingController get _effectiveController =>
+      widget.controller ?? _controller!.value;
+
+  @override
+  void initState() {
+    super.initState();
+    if (widget.controller == null) {
+      _createLocalController();
+    }
+  }
+
+  @override
+  void didUpdateWidget(CupertinoSearchTextField oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.controller == null && oldWidget.controller != null) {
+      _createLocalController(oldWidget.controller!.value);
+    } else if (widget.controller != null && oldWidget.controller == null) {
+      unregisterFromRestoration(_controller!);
+      _controller!.dispose();
+      _controller = null;
+    }
+  }
+
+  @override
+  void restoreState(RestorationBucket? oldBucket, bool initialRestore) {
+    if (_controller != null) {
+      _registerController();
+    }
+  }
+
+  void _registerController() {
+    assert(_controller != null);
+    registerForRestoration(_controller!, 'controller');
+  }
+
+  void _createLocalController([TextEditingValue? value]) {
+    assert(_controller == null);
+    _controller = value == null
+        ? RestorableTextEditingController()
+        : RestorableTextEditingController.fromValue(value);
+    if (!restorePending) {
+      _registerController();
+    }
+  }
+
+  @override
+  String? get restorationId => widget.restorationId;
+
+  VoidCallback _getDefaultOnSuffixTap() {
```

Comment:
nit: you can define the inner function as an instance method (i.e. `void _defaultOnSuffixTap()`), and use it as a tear-off (at line 381):

```dart
        onPressed: widget.onSuffixTap ?? _defaultOnSuffixTap,
```